packages/lit-html/src/lib/lit-html.ts

```diff
@@ -248,6 +246,17 @@ export abstract class Directive {
   update(_part: Part, props: Array<unknown>): unknown {
     return this.render(...props);
   }
+
+  /**
+   * An optional lifecycle method that is called when a directive, or the
+   * template it's attached to, is disconnected from the render tree.
+   *
+   * Note: this is different from being connected to or disconnected from
+   * the document. disconnectedCallback is called even when a directive is
+   * disconnected from a render tree that itself is not connected to the
+   * document.
+   */
+  disconnectedCallback?(): void;
```

Comment:
Consider if we want to name this `disconnected` because as noted in the comments it's not strictly the same as an element's `disconnectedCallback`